### ISSUE TITLE
Registrations fails for non-admin users

### ISSUE THREAD

**Issue by losolio** [losol/eventuras#649]
**Describe the bug**
Non-admin users fails to register for events. 

**To Reproduce**
1. Register for event on frontend as non-admin users
2. Registration fails with a 403 error

**Expected behavior**
Succesful registration

**Error message** 
```
2023-10-24T21:38:19.421215842Z info: Eventuras.Services.Registrations.RegistrationManagementService[0]
2023-10-24T21:38:19.421261043Z       Retrieved event info for EventInfoId 160, Status RegistrationsOpen
2023-10-24T21:38:19.440540234Z info: Eventuras.Services.Registrations.RegistrationManagementService[0]
2023-10-24T21:38:19.440570034Z       Checking create access for registration: UserId 8196c56f-7919-448e-820c-604787002716, EventInfoId 160
2023-10-24T21:38:19.445275605Z warn: Eventuras.Services.Registrations.RegistrationAccessControlService[0]
2023-10-24T21:38:19.445302106Z       User google-oauth2|100284600188326929323 cannot create registration for event 160 and user 8196c56f-7919-448e-820c-604787002716. Access denied.
2023-10-24T21:38:19.448475654Z warn: Eventuras.WebApi.HttpResponseExceptionFilter[0]
2023-10-24T21:38:19.448495754Z       Exception of type Eventuras.Services.Exceptions.NotAccessibleException was handled, resulted in: Microsoft.AspNetCore.Mvc.ForbidResult
```

**Additional context**
It seems like this is comparing an id from the jwt claim (which is not a local user) with the applicationUser id: 
```cs
       private async Task<bool> CheckOwnerOrAdminAccessAsync(ClaimsPrincipal user, Registration registration, CancellationToken cancellationToken)
        {
            if (user.IsAnonymous()) { return false; }
            if (registration.UserId == user.GetUserId()) { return true; }
```

Perhaps we could do something like
```cs 
private async Task<bool> CheckOwnerOrAdminAccessAsync(ClaimsPrincipal user, Registration registration, CancellationToken cancellationToken)
        {
            if (user.IsAnonymous()) { return false; }
            var localUser = await _userRetrievalService.GetUserByEmailAsync(user.GetEmail());
            if (registration.UserId == localUser.Id) { return true; }
```

**Comment by losolio** [comment]
As the v3 API backend has never been used before, I suspect all places where user.GetUserId is used is not working.. Feel free to consider generic ways of getting the local ApplicationUser, and not the userid claim from the jwt.

**Comment by losolio** [comment]
closed by #650
